packages/components/src/public/QueryModel/CustomizeGridViewModal.spec.tsx

```diff
@@ -573,3 +574,38 @@ describe('ColumnChoiceGroup', () => {
         wrapper.unmount();
     });
 });
+
+describe('includedColumnsForCustomizationFilter', () => {
+    test('hidden', () => {
+        let col = QueryColumn.create({ name: 'testColumn', hidden: false });
+        expect(includedColumnsForCustomizationFilter(col, false)).toBeTruthy;
+
+        col = QueryColumn.create({ name: 'testColumn', hidden: true });
+        expect(includedColumnsForCustomizationFilter(col, false)).toBeFalsy();
+
+        col = QueryColumn.create({ name: 'testColumn', hidden: false });
+        expect(includedColumnsForCustomizationFilter(col, true)).toBeTruthy;
+
+        col = QueryColumn.create({ name: 'testColumn', hidden: true });
+        expect(includedColumnsForCustomizationFilter(col, true)).toBeTruthy;
+    });
+
+    test('removeFromViews', () => {
+        let col = QueryColumn.create({ name: 'testColumn', removeFromViews: false });
+        expect(includedColumnsForCustomizationFilter(col, false)).toBeTruthy();
+
+        col = QueryColumn.create({ name: 'testColumn', removeFromViews: true });
+        expect(includedColumnsForCustomizationFilter(col, false)).toBeFalsy();
+    });
+
+    test('removeFromViewCustomization', () => {
+        let col = QueryColumn.create({ name: 'testColumn', removeFromViewCustomization: false });
+        expect(includedColumnsForCustomizationFilter(col, false)).toBeTruthy();
+
+        col = QueryColumn.create({ name: 'testColumn', removeFromViewCustomization: true });
+        expect(includedColumnsForCustomizationFilter(col, false)).toBeFalsy();
```

Comment:
Might be worth adding the `showAllColumns: true` case too

Reply:
done